shopinvader_api_payment_provider_custom/routers/payment_custom.py

```diff
@@ -0,0 +1,70 @@
+# Copyright 2024 ACSONE SA (https://acsone.eu).
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+from typing import Annotated, Any
+from urllib.parse import urljoin
+
+from fastapi import Depends, HTTPException
+
+from odoo import api, models
+
+from odoo.addons.fastapi.dependencies import odoo_env
+from odoo.addons.payment.models.payment_transaction import PaymentTransaction
+from odoo.addons.shopinvader_api_payment.routers import payment_router
+from odoo.addons.shopinvader_api_payment.routers.utils import Payable
+from odoo.addons.shopinvader_api_payment.schemas import TransactionProcessingValues
+
+_logger = logging.getLogger(__name__)
+
+
+@payment_router.post("/payment/providers/custom/pending")
+def custom_payment_pending_msg(
+    payable: str,
+    transaction_id: int,
+    odoo_env: Annotated[api.Environment, Depends(odoo_env)],
+):
+    try:
+        Payable.decode(odoo_env, payable)
+    except Exception as e:
+        _logger.info("Could not decode payable")
+        raise HTTPException(403) from e
+
+    tx_sudo = odoo_env["payment.transaction"].sudo().browse(transaction_id)
+    # TODO: since we get the transaction from its ID, we should check that
+    # the transaction corresponds to the payable. How?
```

Comment:
How to check if the payable corresponds to the transaction?
Should we define a generic check in `shopinvader_api_payment` and then override it in `shopinvader_api_payment_cart` for the specific cart case?

Reply:
Was resolved by adding the transaction_id in the payable.